src/views/_shared/GoogleOAuth.jsx

```diff
@@ -0,0 +1,70 @@
+import React, { Component } from "react";
+import GoogleLogin from "react-google-login";
+// import { LOGIN } from "../../graphql-requests/mutations";
+//import { createBrowserHistory } from 'history';
+import { withRouter } from "react-router-dom";
+
+//let history = createBrowserHistory();
+
+class GoogleOAuth extends Component {
+   constructor(props) {
+      super(props);
+      this.state = {};
+   }
+   signup(res) {
+      const googleresponse = {
+         username: res.profileObj.name,
+         first_name: res.profileObj.name,
+         last_name: res.profileObj.familyName,
+         email: res.profileObj.email,
+         token: res.googleId,
+         Image: res.profileObj.imageUrl,
+         ProviderId: "Google",
+      };
+      //debugger;
+   }
+
+   render() {
+      const responseGoogle = response => {
+         console.log(response);
+         let res = response.profileObj;
+         console.log(res);
+         localStorage.setItem("username", res.familyName);
+         localStorage.setItem("first_name", res.familyName);
+         localStorage.setItem("last_name", res.familyName);
+         localStorage.setItem("email", res.email);
+         localStorage.setItem("token", response.googleId);
+         localStorage.setItem("user_id", response.googleId);
```

Comment:
We should be using the User context for this. For persistence, local storage only needs the User ID and token. We can query for the rest of the information in the Dashboard view. Although that may be easier said than done.

Reply:
These have been cleaned up now to just id, username, and token.

Reply:
As far as the Google dashboard, or just generally a oAuth dashboard route instead of the standard dashboard page, the issue is that the standard dashboard pulls user info from the database. We do not have a mutation currently that puts oAuth users in to the database. So, I'm not apart from implementing the signup feature every time an oAuth user logs in is added to the database in order to render the standard dashboard, I'm not sure what we can do. New mutation?